### DISCUSSION TITLE
Filter users AccountEnabled

### DISCUSSION THREAD

**Discussion by FestTestQA** [microsoft-search/pnp-modern-search#1700] [Q&A] [answered]
Hi,

I am new to using this webpart and do  not know much KQL.
When using this webpart in SP, how could I apply a filter that will filter out users where their Account is not enabled?

**Comment by wobba** [comment, marked as answer]
If it's about people search you would need to find a property on the users profile in the SharePoint User Profile which keeps a value of disabled accounts. You can either have a job reading AAD and then setting a prop on the profile, or you can hide those users from the addressbook in Exchange and filter like mentioned in this post https://thewindowsupdate.com/2019/02/22/improved-filtering-of-people-search-results-in-classic-and-custom-experiences/ using `SPS-HideFromAddressLists`.

**Reply by FestTestQA** [reply]
Many thanks for your time.

**Comment by adumith** [comment]
Hello, 

I have this configuration and for some reason within the results are still showing user accounts that are no longer in the company.

Any idea?

Thanks

{inputQueryText} OfficeNumber:{verticals.value} "SPS-HideFromAddressLists"=0 owstaxIdSPShDepartment:"#8ed8c9ea-7052-4c1d-a4d7-b9c10bffea6f"

**Comment by kasperbolarsen** [comment]
Sure. Most likely the accounts haven't been synced to SharePoint after the SPS-HideFromAddessLists has be set to true. You can check that using one of the tool recommended here, https://microsoft-search.github.io/pnp-modern-search/using-query-tools-to-verify-issues/

**Reply by adumith** [reply]
Hello,

Thank you very much for your response. I have resumed this project and am facing two challenges:

1. When I set that property to true, the search stops working and returns no results. It only works if I set it to 0 or 1. Setting it to 1 should show only active accounts, but it is showing only inactive accounts. It seems to be working in reverse.

2. When configuring the vertical filter, the Department field does not appear in the list of available filters, even though it is properly defined in the schema.

![image](https://github.com/user-attachments/assets/d8d299af-7a7d-490a-8f2e-cc3b06c0455e)

Any ideas how to address this?

Thank you for your help.

**Reply by kasperbolarsen** [reply]
"SPS-HIDEFROMADDRESSLISTS":1 will show those accounts that should not be shown in the Global Address List in Exchange,  that is why we have to negate it,  so it will be -SPS-HIDEFROMADDRESSLISTS":1

Have you checked that Department has a checkmark in Selected Properties?

**Reply by adumith** [reply]
I've created two aliases to try to get the department and employee title. I guess I'll need to reindex the site for this to take effect. What do you think?

![image](https://github.com/user-attachments/assets/304aae35-d0f8-422e-99ed-61cbcf12b2f0)

Thank you so much for your help

**Reply by kasperbolarsen** [reply]
I haven't checked but I think both JobTitle and Department are both sortable and refinable out of the box as they are provided preconfigured when the tenant is created. 
So I don't think you have to do that mapping

**Comment by adumith** [comment]
Thank you so much, 

That's the point, the field is not listed into the Selected Properties combobox:
![image](https://github.com/user-attachments/assets/e56f9d31-e2f5-4c2f-ac74-98d930f81717)

**Comment by kasperbolarsen** [comment]
If it isn't in the list you can always type it in yourself

**Reply by adumith** [reply]
Thanks a lot, Kasper!

It seemed almost too obvious to be true. :)

In the meantime, I found an error on our end: there were user accounts that, despite being disabled, still had the property set to TRUE. Thanks to PowerShell, I was able to identify and fix them.

So we can consider this case resolved.

I’m continuing to explore this powerful people search tool and I’m sure I’ll come back with another challenge.

For anyone who needs the scripts I used, here they are:

The first script checks for disabled accounts where the property is either empty or set to FALSE, and the second script finds those accounts and applies the change.

Import-Module ActiveDirectory
$csvPath = "DisabledUsersReport.csv"
$disabledUsers = Get-ADUser -Filter {Enabled -eq $false} -Properties SamAccountName, DisplayName, EmailAddress, msExchHideFromAddressLists, LastLogonDate, Description
$reportData = $disabledUsers | Select-Object SamAccountName, DisplayName, EmailAddress, msExchHideFromAddressLists, LastLogonDate, Description
$reportData | Export-Csv -Path $csvPath -NoTypeInformation
Write-Host "Reporte generado exitosamente y guardado en $csvPath"


Import-Module ActiveDirectory
$csvPath = "ModifiedDisabledUsersReport.csv"
$disabledUsers = Get-ADUser -Filter {Enabled -eq $false} -Properties SamAccountName, DisplayName, EmailAddress, msExchHideFromAddressLists
$modifiedUsers = @()
foreach ($user in $disabledUsers) {
    if ($user.msExchHideFromAddressLists -ne $true) {
        Set-ADUser -Identity $user.SamAccountName -Replace @{msExchHideFromAddressLists=$true}
        $modifiedUsers += $user
    }
}
$modifiedUsers | Select-Object SamAccountName, DisplayName, EmailAddress, msExchHideFromAddressLists | Export-Csv -Path $csvPath -NoTypeInformation
Write-Host "Las cuentas deshabilitadas han sido revisadas y actualizadas según sea necesario."
Write-Host "Reporte generado exitosamente y guardado en $csvPath"


Thanks again!
